code/go/0chain.net/blobbercore/reference/ref.go

```diff
@@ -225,36 +223,38 @@ func GetReference(ctx context.Context, allocationID, path string) (*Ref, error)
 	return nil, err
 }
 
-func GetReferenceFromLookupHash(ctx context.Context, allocationID, path_hash string) (*Ref, error) {
+// GetLimitedRefFieldsByPath get FileRef selected fields with allocationID and path from postgres
+func GetLimitedRefFieldsByPath(ctx context.Context, allocationID, path string, selectedFields []string) (*Ref, error) {
 	ref := &Ref{}
 	db := datastore.GetStore().GetTransaction(ctx)
-	err := db.Where(&Ref{AllocationID: allocationID, LookupHash: path_hash}).First(ref).Error
+	db = db.Select(selectedFields)
+	err := db.Where(&Ref{AllocationID: allocationID, Path: path}).First(ref).Error
 	if err == nil {
 		return ref, nil
 	}
 	return nil, err
 }
 
-// GetRefType Select type from ref and return it
-func GetRefType(ctx context.Context, allocationID, path string) (string, error) {
-	ref := new(Ref)
+// GetLimitedRefFieldsByLookupHash get FileRef selected fields with allocationID and lookupHash from postgres
+func GetLimitedRefFieldsByLookupHash(ctx context.Context, allocationID, lookupHash string, selectedFields []string) (*Ref, error) {
+	ref := &Ref{}
 	db := datastore.GetStore().GetTransaction(ctx)
-	err := db.Select("type").Where("allocation_id=? AND path=?", allocationID, path).First(ref).Error
-	if err != nil {
-		return "", err
+	db = db.Select(selectedFields)
+	err := db.Where(&Ref{AllocationID: allocationID, LookupHash: lookupHash}).First(ref).Error
+	if err == nil {
+		return ref, nil
```

Comment:
Please handle the error in the Go idiomatic way, i.e. check if `err != nil` to return error early